clang/include/clang/Basic/Attr.td

```diff
@@ -158,7 +158,8 @@ def FunctionTmpl
 
 def HLSLEntry
     : SubsetSubject<Function,
-                    [{S->isExternallyVisible() && !isa<CXXMethodDecl>(S)}],
+                   [{S->getDeclContext()->getRedeclContext()->isFileContext() &&
+                    S->getStorageClass() != SC_Static}],
```

Comment:
Perhaps this is what's expected to enforce that entry functions not be static? From [this description](https://github.com/llvm/llvm-project/blob/a737b8704c031310460d492cef90eee5054cabd7/clang/include/clang/Basic/Attr.td#L84), it sounds to me like it would prevent a static function from being considered an entry function, which is appropriate, but doesn't produce any error when it does.

Reply:
See above.